### PR TITLE
[FIX] [16.0] sale_report_delivered: Consider 0.0 Value valuation layers

### PR DESCRIPTION
**Pull request by Shide**
Take in consideration valuation layers when quantity delivered exceedes demanded quantity.

# Why I've changed the subquery line:
Use COALESCE in subselect query to avoid COALESCE in select query

# Before
https://www.loom.com/share/4c1b1ef838884b3c8dd3c5a6eed9368b?sid=a7524baa-9feb-46d9-b7a0-4d8f687177a8

MT-6136 @moduon @rafaelbn @Gelojr  @fcvalgar  please review if you want :)

### CODE REVIEW

**Review by Gelojr — APPROVED** [review]
Nice job @Shide LGTM
https://www.loom.com/share/cd1045c589f141a084906a9fab0d52fd?sid=3f3f9258-e0f4-4240-84ff-e0a9ab9736c8

**Review by rafaelbn — APPROVED** [review]
Talked, analyzed and tested. Also watched you videos, both! 

Now calculations looks like correct!!! ☺️🤗

Thank you!
❤️

**Comment by Shide** [comment]
I would love to have a review of @sergio-teruel before merging

**Comment by rafaelbn** [comment]
Hello @sergio-teruel ! Good morning 🌅!

We appreciate and consider your opinion here as you are maintainer 🤗

> Use COALESCE in subselect query to avoid COALESCE in select query

We have here a BUG and this PR fix it in several scenarios with diferent warehouse configuration, so it looks like more resilient functionality 

Let us know!

Thank you! 🙏😃❤️

**Review by fcvalgar — APPROVED** [review]
Functional review.

LGTM, thank you @Shide

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Pull request merged**
